ui/src/ElasticSearch.elm

```diff
@@ -7,75 +7,179 @@ import Json.Encode as Encode
 import Types exposing (..)
 
 
-encodeMustClause : Filters -> Encode.Value
-encodeMustClause { product, channel, locale, version, platform, buildId } =
-    let
-        encodeFilter ( name, value ) =
-            Encode.object
-                [ ( if name == "build.id" then
-                        "prefix"
-                    else
-                        "match"
-                  , Encode.object [ ( name, Encode.string value ) ]
-                  )
-                ]
+type ClauseKind
+    = Match
+    | Prefix
+    | Term
 
-        refineFilters ( k, v ) acc =
-            if k == "build.id" then
-                if v /= "" then
-                    ( "build.id", v ) :: acc
-                else
-                    acc
-            else if v /= "all" then
-                ( k, v ) :: acc
-            else
-                acc
-    in
-        [ ( "build.id", buildId )
-        , ( "source.product", product )
-        , ( "target.channel", channel )
-        , ( "target.locale", locale )
-        , ( "target.version", version )
-        , ( "target.platform", platform )
+
+clauseName : ClauseKind -> String
+clauseName kind =
+    case kind of
+        Prefix ->
+            "prefix"
+
+        Match ->
+            "match"
+
+        _ ->
```

Comment:
Why not just use `Term` here? That way you'll get a heads-up if you add a new `ClauseKind`.